File: java/2295_Replace_Elements_in_an_Array/2295_Replace_Elements_in_an_Array.java
class Solution {
    public int[] arrayChange(int[] nums, int[][] operations) {
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            map.put(nums[i], i);
        }
        for (int i = 0; i < operations.length; i++) {
            int origpos = map.get(operations[i][0]);
            map.remove(operations[i][0]);
            nums[origpos] = operations[i][1];
            map.put(operations[i][1], origpos);
        }
        return nums;
    }
}

File: go/2295_Replace_Elements_in_an_Array/2295_Replace_Elements_in_an_Array.go
func arrayChange(nums []int, operations [][]int) []int {
    m := make(map[int]int, len(nums))
    for i, v := range nums {
        m[v] = i // set original position
    }
    for i := 0; i < len(operations); i++ {
        origpos := m[operations[i][0]]
        nums[origpos] = operations[i][1]
        delete(m, operations[i][0]) // remove old number & position pair
        m[operations[i][1]] = origpos
    }
    return nums
}
